FrontEnd/src/ProfilePage/UserProfile.jsx

```diff
@@ -17,15 +21,37 @@ function UserProfilePage({ userUID, userData }) {
   const [isFormValid, setIsFormValid] = useState(false);
   const [serviceCategories, setSelectedCategories] = useState([]);
   const [availableCategories] = useState(CATEGORIES);
-  const [isDropdownVisible, setDropdownVisible] = useState(false);
-  const dropdownRef = useRef(null);
+  const [isSignOutDropdownVisible, setisSignOutDropdownVisible] =
+    useState(false);
+  const [isDeletePostDropdownVisible, setisDeletePostDropdownVisible] =
+    useState(false);
+  const signOutDropdownRef = useRef(null);
+  const deletePostDropdownRef = useRef({});
 
-  const toggleDropdown = () => {
-    setDropdownVisible(!isDropdownVisible);
+  const toggleSignOutDropdown = () => {
+    setisSignOutDropdownVisible(!isSignOutDropdownVisible);
   };
+  const toggleDeletePostDropdown = (index) => {
+    setisDeletePostDropdownVisible(
+      isDeletePostDropdownVisible === index ? null : index
```

Comment:
Why is `setisDeletePostDropdownVisible ` being set with null or index? Shouldn't it be false or true based on the name of the variable and setter?

Reply:
There are multiple posts that are mapped, that's how I am able to differentiate which one of them to delete